rust/foxglove/src/encode.rs

```diff
@@ -80,6 +80,11 @@ impl<T: Encode> TypedChannel<T> {
     pub fn topic(&self) -> &str {
         &self.inner.topic
     }
+
+    /// Returns the channel ID.
+    pub fn id(&self) -> ChannelId {
```

Comment:
If this is only needed for clients, do we want it as part of the public API? Or can this be behind some feature?